golang/cosmos/x/vpurse/vpurse.go

```diff
@@ -22,12 +23,53 @@ type portMessage struct { // comes from swingset's vat-bank
 	Amount    string `json:"amount"`
 }
 
-func NewPortHandler(keeper Keeper) portHandler {
+func NewPortHandler(am AppModule, keeper Keeper) portHandler {
 	return portHandler{
+		am:     am,
 		keeper: keeper,
 	}
 }
 
+type vpurseSingleBalanceUpdate struct {
+	Address string `json:"address"`
+	Denom   string `json:"denom"`
+	Amount  string `json:"amount"`
+}
+
+type vpurseBalanceUpdate struct {
+	Nonce   uint64                      `json:"nonce"`
+	Type    string                      `json:"type"`
+	Updated []vpurseSingleBalanceUpdate `json:"updated"`
+}
+
+var nonce uint64
+
+func marshalBalanceUpdate(ctx sdk.Context, addressToBalance map[string]sdk.Coins) ([]byte, error) {
```

Comment:
Parameter `ctx` not used - drop or add a comment explaining why it's listed.

Reply:
Dropped!